Parser/pegen/pegen.c

```diff
@@ -911,6 +911,48 @@ _PyPegen_number_token(Parser *p)
                     p->arena);
 }
 
+static int // bool
+newline_in_string(Parser *p, const char *cur)
+{
+    for (char c = *cur; cur >= p->tok->buf; c = *--cur) {
+        if (c == '\'' || c == '"') {
+            return 1;
+        }
+    }
+    return 0;
+}
+
+static int // bool
+bad_single_statement(Parser *p)
```

Comment:
Could you also add the comment that is in `parsetok`? I find the explanation useful to read the code having a previous expectation. I am referring to this comment:

```
        /* Check that the source for a single input statement really
            is a single statement by looking at what is left in the
            buffer after parsing.  Trailing whitespace and comments
            are OK.  */
```

Reply:
Done!